File: src/main/java/pl/training/patterns/behavioral/iterator/Application.java
package pl.training.patterns.behavioral.iterator;

import java.util.List;

public class Application {

    public static void main(String[] args) {
        var collection = List.of(1, 2, 3, 4, 5);
        var iterator = collection.iterator();
        //------------------------------------------------------
        while (iterator.hasNext()) {
            System.out.println(iterator.next());
        }
    }

}

File: csharp/src/Training.Patterns/Behavioral/Iterator/Application.cs
namespace Training.Patterns.Behavioral.Iterator;

public class Application
{
    public static void Run()
    {
        IReadOnlyList<int> collection = [1, 2, 3, 4, 5];
        using var iterator = collection.GetEnumerator();
        //------------------------------------------------------
        while (iterator.MoveNext())
        {
            Console.WriteLine(iterator.Current);
        }
    }
}
